### PR TITLE
Update list of boostrap nodes directly from nodes.tox.chat

### PR DESCRIPTION
**Pull request by sudden6**
- [ ] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

This should help us connect to the network faster in case the list changes.

Integration in the toxcore bootstrap process is currently missing.

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5425)

### CODE REVIEW

**Comment by sudden6** [comment]
Ideally we would extend this, so that we can remove the compiled in nodes list and instead compile in a json file which follows the format from nodes.tox.chat.

File: src/net/bootstrapnodeupdater.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+bool BootstrapNodeUpdater::requestBootstrapNodes()
+{
+    nam.setProxy(proxy);
+    connect(&nam, &QNetworkAccessManager::finished, this, &BootstrapNodeUpdater::onRequestComplete);
+
+    QNetworkRequest request{NodeListAddress};
+    request.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
+
+    nam.get(request);
+    return true;
```
Comment (Diadlo) [inline src/net/bootstrapnodeupdater.cpp]: Function returns only one value. Maybe make it `void`?

File: src/net/bootstrapnodeupdater.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+void BootstrapNodeUpdater::onRequestComplete(QNetworkReply* reply)
+{
+    if(reply->error() != QNetworkReply::NoError){
```
Comment (Diadlo) [inline src/net/bootstrapnodeupdater.cpp]: Add space after `if` and `for` for all PR

File: test/net/bsu_test.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+void TestBootstrapNodesUpdater::testOnline()
+{
+    QNetworkProxy proxy{QNetworkProxy::ProxyType::NoProxy};
+    BootstrapNodeUpdater updater{proxy};
+    QSignalSpy spy(&updater, &BootstrapNodeUpdater::availableBootstrapNodes);
+
+    updater.requestBootstrapNodes();
+
+    spy.wait();
+    QCOMPARE(spy.count(), 1); // make sure the signal was emitted exactly one time
```
Comment (Diadlo) [inline test/net/bsu_test.cpp:53]: Maybe also check, that result is is not empty?

Comment (sudden6) [reply test/net/bsu_test.cpp:53]: Done

File: src/net/bootstrapnodeupdater.cpp (hunk 2 above)
Comment (sudden6) [reply src/net/bootstrapnodeupdater.cpp]: Done

File: src/net/bootstrapnodeupdater.cpp (hunk 1 above)
Comment (sudden6) [reply src/net/bootstrapnodeupdater.cpp]: Done

**Review by Diadlo — APPROVED** [review]
:lgtm_strong:

> Reviewed 1 of 12 files at r1, 4 of 11 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5425#-:-LRjMrlCC6g5nZlhVD2G:b-jbsf8z)* status: :shipit: complete! 1 of 1 LGTMs obtained

**Pull request merged**